### PR TITLE
Issue38

### PR DESCRIPTION
**Pull request by jusasiiv**
#38 

Verbose error handling added

### CODE REVIEW

File: controllers/front/validation.php [hunk 1]
```diff
@@ -60,12 +60,50 @@ public function postProcess()
...
+                // New address gen has thrown an error
+                $error_code = $responseObj->message;
+
+                switch ($error_code) {
+                    case "Could not find matching xpub":
+                        $error_str = '<h4>Unable to generate bitcoin address.</h4> <p>Note for site webmaster: There is a problem in the Callback URL. Make sure that you have set your Callback URL from the PrestaShop admin Blockonomics module configuration to your Merchants > Settings.</p>';
+                        break;
+
+                    default:
+                        $error_str = '<h4>Unable to generate bitcoin address.</h4><p> Note for site webmaster: Your webhost is blocking outgoing HTTPS connections. Blockonomics requires an outgoing HTTPS POST (port 443) to generate new address. Check with your webhosting provider to allow this. If problem still persists contact webmaster at blockonomics.co</p>';
```
Comment (blockonomics) [inline controllers/front/validation.php]: Just set _$error_str = $responseObj->message_ Here

File: controllers/front/validation.php [hunk 2]
```diff
@@ -60,12 +60,50 @@ public function postProcess()
...
+                }
+
+                echo $error_str . '<p> If problem persists, please consult this troubleshooting article: <a href="https://blockonomics.freshdesk.com/support/solutions/articles/33000215104-troubleshooting-unable-to-generate-new-address" target="_blank">https://blockonomics.freshdesk.com/support/solutions/articles/33000215104-troubleshooting-unable-to-generate-new-address</a></p>';
+                die();
+            }
+        } elseif(!ini_get('allow_url_fopen')) {
+            // allow_url_fopen not enabled
+            $error_str = '<h4>Unable to generate bitcoin address.</h4><p> Note for site webmaster: Your webhost is blocking outgoing HTTPS connections. Blockonomics requires an outgoing HTTPS POST (port 443) to generate new address. </p><p> If problem persists, please consult this troubleshooting article: <a href="https://blockonomics.freshdesk.com/support/solutions/articles/33000215104-troubleshooting-unable-to-generate-new-address" target="_blank">https://blockonomics.freshdesk.com/support/solutions/articles/33000215104-troubleshooting-unable-to-generate-new-address</a></p>';
```
Comment (blockonomics) [inline controllers/front/validation.php]: This is ask to check fopen and not point to 443 error

**Comment by blockonomics** [comment]
- There is lot of code repetition particular of this `If problem persists, please consult this troubleshooting article: <a href="https://blockonomics.freshdesk.com/support/solutions/articles/33000215104-troubleshooting-unable-to-generate-new-address" target="_blank">https://blockonomics.freshdesk.com/support/solutions/articles/33000215104-troubleshooting-unable-to-generate-new-address</a></p>';`. Refactor code to handle this better
One more case statement required 
   - _This require you to add an xpub in your wallet watcher_ : User added Bitcoin address instead of xpub or didn't add anything
  -

**Comment by shivaenigma** [comment]
Tested looks good. Few changes:
- _Wallet Wathcer > Address Wathcer_  : Spelling
- If problem persists, please consult this troubleshooting article. Hyperlink _this troubleshooting article_

**Pull request merged**